### PR TITLE
Add @ConnectorAttribute

### PR DESCRIPTION
**Pull request by cescoffier**
Defines the @ConnectorAttribute and @ConnectorAttributes annotations

### CODE REVIEW

File: api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/ConnectorAttribute.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) 2018, 2019 Contributors to the Eclipse Foundation
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/ConnectorAttribute.java:2]: 2021

File: api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/ConnectorAttributes.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) 2018, 2019 Contributors to the Eclipse Foundation
```
Comment (Emily-Jiang) [inline api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/ConnectorAttributes.java:2]: 2021

File: api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/ConnectorAttribute.java (hunk 1 above)
Comment (cescoffier) [reply api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/ConnectorAttribute.java]: oups... Fixed

File: api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/ConnectorAttributes.java (hunk 2 above)
Comment (cescoffier) [reply api/src/main/java/org/eclipse/microprofile/reactive/messaging/spi/ConnectorAttributes.java]: fixed

File: spec/src/main/asciidoc/architecture.asciidoc [hunk 3]
```diff
@@ -1295,6 +1295,31 @@ The configuration passed to the `IncomingConnectorFactory` and `OutgoingConnecto
 * `channel-name` attribute indicating the name of the channel being configured,
 * `connector` attribute must match the name given to the `@Connector` qualifier.
 
+==== Connector attribute
+
+To help tools (IDEs, documentation generator) to extract the configuration of each connector, the specification recommends the usage of the `org.eclipse.microprofile.reactive.messaging.spi.ConnectorAttribute` annotation.
```
Comment (Emily-Jiang) [inline spec/src/main/asciidoc/architecture.asciidoc:1300]: I am wondering whether we could add an optional TCK. What do you think?

Comment (cescoffier) [reply spec/src/main/asciidoc/architecture.asciidoc:1300]: I don't believe we can, what would it be? 
The implementations do not do anything with these attributes.

Comment (Emily-Jiang) [reply spec/src/main/asciidoc/architecture.asciidoc:1300]: okay, I have not thought of a solution as yet, but I thought you might know. If you don't believe we could, it is okay.

Comment (Emily-Jiang) [reply spec/src/main/asciidoc/architecture.asciidoc:1300]: To help tools (IDEs, documentation generator) to extract the configuration of each connector, the specification recommends the usage of the `org.eclipse.microprofile.reactive.messaging.spi.ConnectorAttribute` annotation. =>

To help tools (IDEs, documentation generator) to extract the configuration of each connector, the specification provides the `org.eclipse.microprofile.reactive.messaging.spi.ConnectorAttribute` annotation for implementations to create a good ecosystem with the tools. However, the support for this SPI is not mandatory.

Comment (cescoffier) [reply spec/src/main/asciidoc/architecture.asciidoc]: Fixed.

**Review by Emily-Jiang — APPROVED** [review]

**Pull request merged**